File: app/src/main/java/za/org/grassroot2/dagger/activity/ActivityComponent.java
package za.org.grassroot2.dagger.activity;

import dagger.Subcomponent;
import za.org.grassroot2.view.activity.CreateActionActivity;
import za.org.grassroot2.view.activity.GrassrootActivity;
import za.org.grassroot2.view.activity.GroupDetailsActivity;
import za.org.grassroot2.view.activity.LoginActivity;
import za.org.grassroot2.view.activity.LoginActivity2;
import za.org.grassroot2.view.activity.PickContactActivity;
import za.org.grassroot2.view.activity.RegisterActivity;
import za.org.grassroot2.view.activity.WelcomeActivity;
import za.org.grassroot2.view.fragment.GrassrootFragment;
import za.org.grassroot2.view.fragment.GroupSelectionFragment;
import za.org.grassroot2.view.fragment.GroupTasksFragment;
import za.org.grassroot2.view.fragment.GroupsFragment;
import za.org.grassroot2.view.fragment.ItemCreatedFragment;
import za.org.grassroot2.view.fragment.ItemSelectionFragment;
import za.org.grassroot2.view.fragment.MeetingDateFragment;
import za.org.grassroot2.view.fragment.SingleTextMultiButtonFragment;

/**
 * Created by luke on 2017/08/08.
 */
@PerActivity
@Subcomponent(modules = {ActivityModule.class})
public interface ActivityComponent {

    void inject(SingleTextMultiButtonFragment fragment);
    void inject(ItemSelectionFragment fragment);
    void inject(GrassrootActivity target);
    void inject(LoginActivity target);

    void inject(LoginActivity2 target);
    void inject(GroupsFragment fragment);
    void inject(GroupDetailsActivity target);
    void inject(GroupTasksFragment fragment);
    void inject(GrassrootFragment grassrootFragment);
    void inject(PickContactActivity activity);
    void inject(GroupSelectionFragment fragment);
    void inject(MeetingDateFragment fragment);
    void inject(CreateActionActivity activity);
    void inject(ItemCreatedFragment fragment);

    void inject(WelcomeActivity target);

    void inject(RegisterActivity target);
}

File: app/src/main/java/za/org/grassroot2/dagger/activity/ActivityComponent.kt
package za.org.grassroot2.dagger.activity

import dagger.Subcomponent
import za.org.grassroot2.dagger.fragment.FragmentComponent
import za.org.grassroot2.dagger.fragment.FragmentModule
import za.org.grassroot2.view.activity.*
import za.org.grassroot2.view.fragment.*

/**
 * Created by luke on 2017/08/08.
 */
@PerActivity
@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

    fun inject(fragment: ItemSelectionFragment)
    fun inject(target: GrassrootActivity)
    fun inject(target: DashboardActivity)
    fun inject(target: LoginActivity)
    fun inject(fragment: GroupsFragment)
    fun inject(target: GroupDetailsActivity)
    fun inject(fragment: GroupTasksFragment)
    fun inject(grassrootFragment: GrassrootFragment)
    fun inject(activity: PickContactActivity)
    fun inject(fragment: GroupSelectionFragment)
    fun inject(fragment: MeetingDateFragment)
    fun inject(activity: CreateActionActivity)
    fun inject(fragment: ItemCreatedFragment)
    fun inject(activity: WelcomeActivity)
    fun inject(activity: RegisterActivity)
    fun inject(activity: ForgotPasswordActivity)
    fun inject(activity: MeetingDetailsActivity)
    fun inject(activity: CreatePostActivity)
    fun plus(module: FragmentModule) : FragmentComponent

}